packages/stark-ui/src/util/dom.util.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Stark specific DOM commands
+ */
+export class StarkDOMUtil {
+	/**
+	 * Recursive function to look for the parent element with the specified tag name
+	 * it will stop the recursion when it reaches the <body> element
+	 * @param element - HTML element whose parent will be searched.
+	 * @param tagName - Tag name of the parent element to be searched.
+	 * @returns The searched element in case it was found, otherwise it returns undefined
+	 */
+	public static searchParentElementByTag(element: Element, tagName: string): Element | undefined {
+		if (element.parentElement && element.parentElement.tagName === tagName.toUpperCase()) {
+			return element.parentElement;
+		} else if (element.parentElement && element.parentElement.tagName !== "BODY") {
+			return StarkDOMUtil.searchParentElementByTag(element.parentElement, tagName);
+		} else {
+			return undefined;
+		}
+	}
+
+	/**
+	 * Recursive function to look for the parent element with the specified class
+	 * it will stop the recursion when it reaches the <body> element
+	 * @param element - HTML element whose parent will be searched
+	 * @param className - Class name of the parent element to be searched
+	 * @returns The searched element in case it was found, otherwise it returns undefined.
+	 */
+	public static searchParentElementByClass(element: Element, className: string): Element | undefined {
+		if (element.parentElement && this.hasClass(element.parentElement, className)) {
```

Comment:
If the `hasClass()` method is public then you should call this method like this: `StarkDOMUtil.hasClass(....)`